src/test.js

```diff
@@ -623,25 +604,61 @@ function skip( testName ) {
 }
 
 // Will be exposed as QUnit.only
-function only( testName, expected, callback, async ) {
+function only( testName, callback ) {
 	var newTest;
 
 	if ( focused )  { return; }
 
 	QUnit.config.queue.length = 0;
 	focused = true;
 
-	if ( arguments.length === 2 ) {
-		callback = expected;
-		expected = null;
-	}
-
 	newTest = new Test( {
 		testName: testName,
-		expected: expected,
-		async: async,
 		callback: callback
 	} );
 
 	newTest.queue();
 }
+
+function internalStop() {
+
+	// If a test is running, adjust its semaphore
+	config.current.semaphore += 1;
+
+	pauseProcessing();
+}
+
+function internalStart() {
+
+	// If a test is running, adjust its semaphore
+	config.current.semaphore -= 1;
+
+	// If semaphore is non-numeric, throw error
+	if ( isNaN( config.current.semaphore ) ) {
```

Comment:
I cannot do any inline comments because it's not close from a change I did:

You can see [here](https://github.com/jquery/qunit/pull/976/commits/21c254ac2f65b4833d80eee055aa47141c737045#diff-f8f3ba9b60ca5c3a0a1dd233f0d34122R47) I have this `resumeProcessing()` that can't be called with the current test where `done` was set in. Doing it breaks the current suite.

We can investigate this in another issue, but for me it looks we are leaking the async functionality somehow and our current tests are returning false positives.